aten/src/ATen/native/mps/operations/RMSNorm.mm

```diff
@@ -0,0 +1,68 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/empty_like.h>
+#include <ATen/ops/rms_norm_native.h>
+#endif
+#include <ATen/mps/MPSProfiler.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <ATen/native/mps/operations/RMSNorm.h>
+#include <fmt/format.h>
+
+namespace at::native::mps {
+
+#ifndef PYTORCH_JIT_COMPILE_SHADERS
+static auto& lib = MetalShaderLibrary::getBundledLibrary();
+#else
+#include <ATen/native/mps/RMSNorm_metallib.h>
+#endif
+
+Tensor rms_norm_mps_kernel(const Tensor& input,
+                           c10::SymIntArrayRef normalized_shape,
+                           const Tensor& weight,
+                           const double eps) {
+  auto output = at::empty_like(input);
+  const int normalized_ndim = normalized_shape.size();
+  const auto input_shape = input.sizes();
+  const auto input_ndim = input.dim();
+  const int axis = input_ndim - normalized_ndim;
+  const size_t M = static_cast<size_t>(c10::multiply_integers(input_shape.cbegin(), input_shape.cbegin() + axis));
+  const size_t N = static_cast<size_t>(c10::multiply_integers(input_shape.cbegin() + axis, input_shape.cend()));
+
+  static constexpr int SIMD_SIZE = 32;
+  static constexpr int N_READS = 4;
+  static constexpr int LOOPED_LIMIT = 4096;
+  const std::string name = N > LOOPED_LIMIT ? "rms_norm_looped" : "rms_norm";
+
+  MPSStream* mpsStream = getCurrentMPSStream();
+
+  dispatch_sync_with_rethrow(mpsStream->queue(), ^() {
+    @autoreleasepool {
+      id<MTLComputeCommandEncoder> computeEncoder = mpsStream->commandEncoder();
+      const std::string kernel = fmt::format("{}_{}", name, scalarToMetalTypeString(output));
+      id<MTLComputePipelineState> rms_norm_pso = lib.getPipelineStateForFunc(kernel);
```

Comment:
Nits
```suggestion
      aut computeEncoder = mpsStream->commandEncoder();
      const auto kernel = fmt::format("{}_{}", name, scalarToMetalTypeString(output));
      auto rms_norm_pso = lib.getPipelineStateForFunc(kernel);
```

Reply:
For someone that is not familiar with Metal, explicit types really help follow the code. I was in that position few months ago. For that reason, I think I will keep them fully spelled out.